packages/queries/src/validators/isQueryParam.ts

```diff
@@ -0,0 +1,9 @@
+import { QueryParam } from '../types/QueryParam';
+
+export function isQueryParam(x: any): x is QueryParam {
+    return (
```

Comment:
I think it's safer if we just do two checks:

1. Check that x is not null or undefined
2. Check that x.type === "synthql::param"

For example: Maybe undefined is a valid value sometimes.